hooks/tk-multi-loader2/tk-3de4_actions.py

```diff
@@ -0,0 +1,289 @@
+            is raised midway through.
+
+        :param list actions: Action dictionaries.
+        """
+        for single_action in actions:
+            name = single_action["name"]
+            sg_publish_data = single_action["sg_publish_data"]
+            params = single_action["params"]
+            self.execute_action(name, params, sg_publish_data)
+
+    def execute_action(self, name, params, sg_publish_data):
+        """
+        Execute a given action. The data sent to this be method will
+        represent one of the actions enumerated by the generate_actions method.
+
+        :param name: Action name string representing one of the items returned by generate_actions.
+        :param params: Params data, as specified by generate_actions.
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :returns: No return value expected.
+        """
+        app = self.parent
+        app.logger.debug(
+            "Execute action called for action %s. "
+            "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data)
+        )
+
+        # resolve path
+        # toolkit uses utf-8 encoded strings internally and Maya API expects unicode
+        # so convert the path to ensure filenames containing complex characters are supported
+        path = self.get_publish_path(sg_publish_data).decode("utf-8")
+
+        if name == "import_image_seq":
+            self._import_image_seq(path, sg_publish_data)
+
+    ##############################################################################################################
+    # helper methods which can be subclassed in custom hooks to fine tune the behaviour of things
+
+    def _import_image_seq(self, path, sg_publish_data):
+        """
+        Import and image sequence and assign it to the selected cameras.
+
+        :param str path: The file path to load.
+        :param dict sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        """
+        app = self.parent
+        path, start, end, step = get_hash_path_and_range_info_from_seq(path)
+        name = app.engine.context.entity["name"]
+
+        camera_count = tde4.getNoCameras()
+        if camera_count:
+            selected_cameras = filter(is_sequence_camera, tde4.getCameraList(True))
+            if not selected_cameras:
+                QtGui.QMessageBox.warning(
+                    None,
+                    "No sequence cameras selected",
+                    "Please select a sequence camera and try again"
+                )
+                return
+            app.logger.info("%d sequence cameras selected, assigning to all", len(selected_cameras))
+            for cam_id in selected_cameras:
+                current_name = tde4.getCameraName(cam_id)
+                app.logger.debug("Current camera: '%s'", current_name)
+                if not current_name.startswith(name):
```

Comment:
Refactor to `if ... else ...` by switching contents of `if not ...` with `else ...`